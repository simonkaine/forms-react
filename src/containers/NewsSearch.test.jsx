import React from 'react';
import NewsSearch from './NewsSearch';
import { render, screen } from '@testing-library/react';

const testArticle = {
    author: "Ryan Lawyer", 
    title: "What does Zillowâ€™s exit tell us about the health of the iBuying market?"
};

describe('Shows news articles', () => {
    it('Should display a list from news articles', async () => {
        // Loading test
        render(<NewsSearch />)
        screen.getByText('LOADING!!!...')

        // snapshot test all presentational components -> article list 'list', 'article' & search
        // 1. on screen find the html match for 'list' & expect list snapshot 
        const articlesList = await screen.findByRole('list', { name: 'list' }) 
        expect(articlesList).toMatchSnapshot();

        // behavior test - on changing search refetch articles

    })
})
