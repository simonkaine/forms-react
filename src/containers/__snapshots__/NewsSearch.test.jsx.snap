// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shows news articles Should display a list from news articles 1`] = `
<ul
  aria-label="list"
>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          An accelerator and fund says it's giving away its voting rights to scientist founders: "They lose control"
        </h2>
        <h3>
          By: 
          Connie Loizos
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Daily Crunch: IBM managed IT services spinoff Kyndryl starts trading on NYSE
        </h2>
        <h3>
          By: 
          Alex Wilhelm
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Theodore Tewksbury to take over as Velodyne CEO
        </h2>
        <h3>
          By: 
          Aria Alamalhodaei
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Mark Cuban and former Oculus CEO back 3D e-commerce startup Vntana
        </h2>
        <h3>
          By: 
          Lucas Matney
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Twitter now lets you tune into Spaces without an account via direct links
        </h2>
        <h3>
          By: 
          Aisha Malik
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Push it jumps to No. 1 on App Store promising a way to send push notifications to friends
        </h2>
        <h3>
          By: 
          Sarah Perez
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          What does Zillowâ€™s exit tell us about the health of the iBuying market?
        </h2>
        <h3>
          By: 
          Ryan Lawler
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Google Cloud's $1B investment in CME Group part of long-term infrastructure services deal
        </h2>
        <h3>
          By: 
          Ron Miller
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Rivian hit with gender discrimination lawsuit that alleges toxic 'bro culture'
        </h2>
        <h3>
          By: 
          Kirsten Korosec
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
  <li>
    <figure>
      <figcaption>
        <h2>
          Article Title: 
          Uber squeaks tiny adjusted profit in Q3 despite $2.4B net loss
        </h2>
        <h3>
          By: 
          Alex Wilhelm
        </h3>
        <hr />
      </figcaption>
    </figure>
  </li>
</ul>
`;
